### PR TITLE
[RSDK-8326] Remove servo Digital Interrupts

### PR DESCRIPTION
**Pull request by randhid**
We are deprecating this type in the Raspberry Pi module, we can keep the config attribute since olivia will be adding new pin type attributes, but we can remove this now.

To Do:
- [x] Test on a pi.

### CODE REVIEW

**Review by oliviamiller — COMMENTED** [review]
🎉 !! There is a test for this in `external_hardware_test.go` that we should remove, and should remove from readme

File: utils/digital_interrupts.go [hunk 1]
```diff
@@ -48,15 +47,7 @@ func CreateDigitalInterrupt(cfg DigitalInterruptConfig) (ReconfigurableDigitalIn
 		cfg.Type = "basic"
```
Comment (susmitaSanyal) [inline utils/digital_interrupts.go:47]: This if statement is invalid now that we only have 'basic' type.

Comment (susmitaSanyal) [reply utils/digital_interrupts.go:47]: I think this is a good time to get rid of cfg.Type.

Comment (randhid) [reply utils/digital_interrupts.go:47]: No, @oliviamiller's just approved scope doc has a "type" config value for pins; there is a refactor that gets rid of basic interrupt altogether, but I am not folding in that work with the current pr.

Comment (randhid) [reply utils/digital_interrupts.go:47]: I'll get rid of the if check though

File: README.md [hunk 2]
```diff
@@ -154,7 +154,7 @@ The following properties are available for `digital_interrupts`:
 | ---- | ---- | --------- | ----------- |
 |`name` | string | **Required** | Your name for the digital interrupt. |
 |`pin`| string | **Required** | The pin number of the board's GPIO pin that you wish to configure the digital interrupt for. |
-|`type`| string | Optional | <ul><li>`basic`: Recommended. Tracks interrupt count. </li> <li>`servo`: For interrupts configured for a pin controlling a [servo](https://docs.viam.com/components/servo/). Tracks pulse width value. </li></ul> |
+|`type`| string | Optional | <ul><li>`basic`: Recommended. Tracks interrupt count. </li> </ul> |
```
Comment (susmitaSanyal) [inline README.md:157]:
```suggestion
|`type`| string | Optional | Default: `basic`: Tracks interrupt count. </li> </ul> |
```

File: utils/digital_interrupts.go [hunk 3]
```diff
@@ -44,19 +43,7 @@ type ReconfigurableDigitalInterrupt interface {
 // CreateDigitalInterrupt is a factory method for creating a specific DigitalInterrupt based
 // on the given config. If no type is specified, a BasicDigitalInterrupt is returned.
```
Comment (susmitaSanyal) [inline utils/digital_interrupts.go:44]:
```suggestion
// CreateDigitalInterrupt is a factory method for creating a BasicDigitalInterrupt type.
```

**Review by susmitaSanyal — APPROVED** [review]
few nits, but LGTM

**Comment by randhid** [comment]
Tested on a pi
```sh
warning: GOPATH set to GOROOT (/home/ada/.local/share/go) has no effect
=== RUN   TestPiPigpio
    daemon.go:51: 2024-10-02T19:41:05.463+0100	INFO		rpi/daemon.go:51	pigpiod is already running
    board.go:129: 2024-10-02T19:41:05.516+0100	INFO		rpi/board.go:129	successfully started pigpiod
    rpi_external_test.go:46: not running as root on a pi
--- SKIP: TestPiPigpio (0.06s)
=== RUN   TestPiServo
=== RUN   TestPiServo/servo_initialize_with_pin_error
=== RUN   TestPiServo/check_new_servo_defaults
=== NAME  TestPiServo
    rpiservo.go:102: 2024-10-02T19:41:05.517+0100	INFO		rpi-servo/rpiservo.go:102	setting default pwm frequency of 50 Hz
=== RUN   TestPiServo/check_set_default_position
=== NAME  TestPiServo
    rpiservo.go:102: 2024-10-02T19:41:05.520+0100	INFO		rpi-servo/rpiservo.go:102	setting default pwm frequency of 50 Hz
--- PASS: TestPiServo (0.00s)
    --- PASS: TestPiServo/servo_initialize_with_pin_error (0.00s)
    --- PASS: TestPiServo/check_new_servo_defaults (0.00s)
    --- PASS: TestPiServo/check_set_default_position (0.00s)
PASS
ok  	raspberry-pi/tests	0.147s
```

**Pull request merged**